---
title: TypeError
slug: Web/JavaScript/Reference/Global_Objects/TypeError
tags:
  - Error
  - JavaScript
  - TypeError
  - помилка
translation_of: Web/JavaScript/Reference/Global_Objects/TypeError
---
<div>{{JSRef}}</div>

<p>Об'єкт <code><strong>TypeError</strong></code> позначає помилку, що виникає при використанні значення невідповідного типу.</p>

<h2 id="Синтаксис">Синтаксис</h2>

<pre class="syntaxbox"><code>new TypeError([<var>message</var>[, <var>fileName</var>[, <var>lineNumber</var>]]])</code></pre>

<h3 id="Параметри">Параметри</h3>

<dl>
 <dt><code>message</code></dt>
 <dd>Необов'язковий. Зрозумілий людині опис помилки.</dd>
 <dt><code>fileName</code> {{non-standard_inline}}</dt>
 <dd>Необов'язковий. Ім'я файлу, код з якого спричинив виняток.</dd>
 <dt><code>lineNumber</code> {{non-standard_inline}}</dt>
 <dd>Необов'язковий. Номер рядка в коді, що спричинив виняток.</dd>
</dl>

<h2 id="Опис">Опис</h2>

<p><code>TypeError</code> викидається, коли:</p>

<ul>
 <li>операнд або аргумент, що передається у функцію, несумісний з типом, який очікується цим оператором чи функцією, або</li>
 <li>при спробі змінити значення, яке не можна змінювати, або</li>
 <li>при спробі використати значення невідповідним чином.</li>
</ul>

<h2 id="Властивості">Властивості</h2>

<dl>
 <dt><code>TypeError.prototype</code></dt>
 <dd>Дозволяє додавати властивості до об'єктів <code>TypeError</code>.</dd>
</dl>

<h2 id="Методи">Методи</h2>

<p>Сам <code>TypeError</code> не має власних методів, але успадковує деякі методи через ланцюжок прототипів.</p>

<h2 id="Екземпляри_TypeError">Екземпляри <code>TypeError</code></h2>

<h3 id="Властивості_2">Властивості</h3>

<div>{{page('/uk/docs/Web/JavaScript/Reference/Global_Objects/TypeError/prototype', 'Властивості')}}</div>

<h3 id="Методи_2">Методи</h3>

<div>{{page('/uk/docs/Web/JavaScript/Reference/Global_Objects/TypeError/prototype', 'Методи')}}</div>

<h2 id="Приклади">Приклади</h2>

<h3 id="Перехоплення_TypeError">Перехоплення <code>TypeError</code></h3>

<pre class="brush: js">try {
  null.f();
} catch (e) {
  console.log(e instanceof TypeError) // true
  console.log(e.message)              // "null has no properties"
  console.log(e.name)                 // "TypeError"
  console.log(e.fileName)             // "Scratchpad/1"
  console.log(e.lineNumber)           // 2
  console.log(e.columnNumber)         // 2
  console.log(e.stack)                // "@Scratchpad/2:2:3\n"
}
</pre>

<h3 id="Створення_TypeError">Створення <code>TypeError</code></h3>

<pre class="brush: js">try {
  throw new TypeError('Привіт', "someFile.js", 10);
} catch (e) {
  console.log(e instanceof TypeError) // true
  console.log(e.message)              // "Привіт"
  console.log(e.name)                 // "TypeError"
  console.log(e.fileName)             // "someFile.js"
  console.log(e.lineNumber)           // 10
  console.log(e.columnNumber)         // 0
  console.log(e.stack)                // "@Scratchpad/2:2:9\n"
}
</pre>

<h2 id="Специфікації">Специфікації</h2>

<table class="standard-table">
 <tbody>
  <tr>
   <th scope="col">Специфікація</th>
  </tr>
  <tr>
   <td>{{SpecName('ESDraft', '#sec-native-error-types-used-in-this-standard-typeerror', 'TypeError')}}</td>
  </tr>
 </tbody>
</table>

<h2 id="Сумісність_з_веб-переглядачами">Сумісність з веб-переглядачами</h2>

<div>


<p>{{Compat("javascript.builtins.TypeError")}}</p>
</div>

<h2 id="Див._також">Див. також</h2>

<ul>
 <li>{{jsxref("Error")}}</li>
</ul>
